modules/plugins/languages/typst.nix

```diff
@@ -150,6 +151,57 @@ in {
           };
         };
       };
+      typst-concealer = {
+        enable = mkEnableOption ''
+          [typst-concealer]: https://github.com/PartyWumpus/typst-concealer
+
+          Inline typst preview for Neovim via [typst-concealer]
+        '';
+
+        mappings = {
+          toggleConcealing = mkMappingOption "Enable typst-concealer in buffer" "<leader>TT";
+        };
+
+        setupOpts = mkPluginSetupOption "typst-concealer" {
+          do_diagnostics = mkOption {
+            type = nullOr bool;
+            default = !cfg.lsp.enable;
+            description = "Should typst-concealer provide diagnostics on error?";
+          };
+          color = mkOption {
+            type = nullOr str;
+            default = null;
+            example = "rgb(\"#f012be\")";
+            description = "What color should typst-concealer render text/stroke with? (only applies when styling_type is 'colorscheme')";
+          };
+          enabled_by_default = mkOption {
+            type = nullOr bool;
+            default = null;
+            description = "Should typst-concealer conceal newly opened buffers by default?";
+          };
+          styling_type = mkOption {
+            type = nullOr (enum ["simple" "none" "colorscheme"]);
+            default = null;
+            description = "What kind of styling should typst-concealer apply to your typst?";
+          };
+          ppi = mkOption {
+            type = nullOr int;
+            default = null;
+            description = "What PPI should typst render at. Plugin default is 300, typst's normal default is 144.";
+          };
+          typst_location = mkOption {
+            type = str;
+            default = getExe pkgs.typst;
+            description = "Where should typst-concealer look for your typst binary?";
+            example = ''lib.getExe pkgs.typst'';
+          };
+          conceal_in_normal = mkOption {
+            type = nullOr bool;
+            default = null;
+            description = "Should typst-concealer still conceal when the normal mode cursor goes over a line.";
+          };
```

Comment:
`null` doesn't actually mean null in the traditional sense here. If you pass null to the default value, it will be converted to Lua's `nil` and that will be passed to your plugin options instead. If you want to use your plugin's defaults, simplfy don't define those options and the freeform module will allow using any value. Alternatively you can match the default values here IF you want to define the options for documentation purposes.

Reply:
No need. I assumed you were confused about the null conversion but seems it's intended. You may discard my comment, I'm going to merge now.

P.S. If you can squash your commits into one I can avoid squash merging :)